### PR TITLE
feat(cli): add cli support for list verb

### PR DESCRIPTION
**Pull request by schoren**
This PR adds cli support for the list verb of the test resource. 

Loom: https://www.loom.com/share/65660cd88b2d4e8ca8827cd3f47ead02

## Fixes

- https://github.com/kubeshop/tracetest/issues/2665

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: cli/cmd/apply_cmd.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ var applyCmd = &cobra.Command{
 		ctx := context.Background()
 
 		analytics.Track("Resource Apply", "cmd", map[string]string{
-			resourceType: resourceType,
+			"resourceType": resourceType,
```
Comment (schoren) [inline cli/cmd/apply_cmd.go:28]: this is not related to this PR directly, but it's a typo

File: server/resourcemanager/resource_manager.go [hunk 2]
```diff
@@ -340,7 +340,7 @@ func (m *manager[T]) list(w http.ResponseWriter, r *http.Request) {
 	}
 
 	listFn := m.rh.List
-	if isRequestForAugmented(r) {
+	if isRequestForAugmented(r) && m.rh.ListAugmented != nil {
```
Comment (schoren) [inline server/resourcemanager/resource_manager.go:343]: this fixes a panic on the server when getting an augmented list to a resource that have not registered a `ListAugmented` method

**Comment by kdhamric** [comment]
Can we add a loom video showing it work (or does that not make sense for this PR?)

File: cli/utils/api.go [hunk 3]
```diff
@@ -225,6 +225,7 @@ func (resourceClient ResourceClient) List(ctx context.Context, listArgs ListArgs
 	}
 
 	if resp.StatusCode == http.StatusBadRequest {
+		// TODO: this function is masking potential errors, add correct error handling
```
Comment (mathnogueira) [inline cli/utils/api.go:228]: Should this be done in this PR or a new one?

File: testing/cli-e2etest/testscenarios/test/list_test_test.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+		require.Equal("http", listTest.Spec.Trigger.Type)
+		require.Equal("http://demo-api:8081/pokemon/import", listTest.Spec.Trigger.HTTPRequest.URL)
+		require.Equal("POST", listTest.Spec.Trigger.HTTPRequest.Method)
+		require.Equal(`'{"id":52}'`, listTest.Spec.Trigger.HTTPRequest.Body)
+		require.Len(listTest.Spec.Trigger.HTTPRequest.Headers, 1)
+		require.Equal("Content-Type", listTest.Spec.Trigger.HTTPRequest.Headers[0].Key)
+		require.Equal("application/json", listTest.Spec.Trigger.HTTPRequest.Headers[0].Value)
+	})
+
+	// 	t.Run("list with JSON format", func(t *testing.T) {
```
Comment (mathnogueira) [inline testing/cli-e2etest/testscenarios/test/list_test_test.go:111]: Why is this commented out? Should it be removed?

**Review by mathnogueira — APPROVED** [review]

**Comment by schoren** [comment]
closing in favor of https://github.com/kubeshop/tracetest/pull/2849

**Pull request closed without merging**